### ISSUE TITLE
[Bug]: [chaos][pulsar]search and insert not work after pulsar pod recovered from pod kill

### ISSUE THREAD

**Issue by zhuwenxing** [milvus-io/milvus#14689]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:master-20211230-95f0e9a
- Deployment mode(standalone or cluster):cluster
- SDK version(e.g. pymilvus v2.0.0rc2): 2.0.0rc9.dev28
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

1. when running `pytest -s -v ../testcases/test_e2e.py`, flush hangs about 9 mins
```
[2021-12-31 14:21:44 - INFO - ci_test]: assert flush: 597.1756601333618, entities: 3000 (test_e2e.py:39)
```
2. when running `pytest -s -v ../testcases/test_e2e.py`, search raise error
```
[2021-12-31 14:21:45 - DEBUG - ci_test]: (api_request)  : [Collection.search] args: [[[0.04211650020944873, 0.1034729685590432, 0.11564181577433845, 0.07797765538635734, 0.02297460176682076, 0.0940157949377532, 0.08561942656825403, 0.09612052836944039, 0.07595184069534243, 0.0668874076220618, 0.0048142831420574465, 0.09033997337646936, 0.11371023241534171, 0.02822739507309657, 0.04......, kwargs: {} (api_request.py:55)
[2021-12-31 14:22:05 - ERROR - pymilvus.client.grpc_handler]: RPC error: _execute_search_requests{'API start': '2021-12-31 14:21:45.655978', 'RPC start': '2021-12-31 14:21:45.656008', 'RPC error': '2021-12-31 14:22:05.660113'} (grpc_handler.py:91)
[2021-12-31 14:22:05 - ERROR - pymilvus.client.grpc_handler]: RPC error: search{'API start': '2021-12-31 14:21:45.630729', 'RPC start': '2021-12-31 14:21:45.630743', 'RPC error': '2021-12-31 14:22:05.660721'} (grpc_handler.py:91)
[2021-12-31 14:22:05 - ERROR - ci_test]: Traceback (most recent call last):
  File "/Users/zilliz/workspace/backup/milvus/tests/python_client/utils/api_request.py", line 22, in inner_wrapper
    res = func(*args, **kwargs)
  File "/Users/zilliz/workspace/backup/milvus/tests/python_client/utils/api_request.py", line 56, in api_request
    return func(*arg, **kwargs)
  File "/Users/zilliz/opt/anaconda3/lib/python3.8/site-packages/pymilvus/orm/collection.py", line 696, in search
    res = conn.search(self._name, data, anns_field, param, limit, expr,
  File "/Users/zilliz/opt/anaconda3/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 92, in handler
    raise e
  File "/Users/zilliz/opt/anaconda3/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 76, in handler
    return func(*args, **kwargs)
  File "/Users/zilliz/opt/anaconda3/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 108, in handler
    return func(self, *args, **kwargs)
  File "/Users/zilliz/opt/anaconda3/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 467, in search
    return self._execute_search_requests(requests, timeout, **_kwargs)
  File "/Users/zilliz/opt/anaconda3/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 92, in handler
    raise e
  File "/Users/zilliz/opt/anaconda3/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 76, in handler
    return func(*args, **kwargs)
  File "/Users/zilliz/opt/anaconda3/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 417, in _execute_search_requests
    raise pre_err
  File "/Users/zilliz/opt/anaconda3/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 405, in _execute_search_requests
    response = ft.result()
  File "/Users/zilliz/opt/anaconda3/lib/python3.8/site-packages/grpc/_channel.py", line 744, in result
    raise self
grpc._channel._MultiThreadedRendezvous: <_MultiThreadedRendezvous of RPC that terminated with:
	status = StatusCode.DEADLINE_EXCEEDED
	details = "Deadline Exceeded"
	debug_error_string = "{"created":"@1640931725.659538000","description":"Error received from peer ipv4:10.96.23.132:19530","file":"src/core/lib/surface/call.cc","file_line":1068,"grpc_message":"Deadline Exceeded","grpc_status":4}"
>
 (api_request.py:35)
[2021-12-31 14:22:05 - ERROR - ci_test]: (api_response) : <_MultiThreadedRendezvous of RPC that terminated with:
	status = StatusCode.DEADLINE_EXCEEDED
	details = "Deadline Exceeded"
	debug_error_string = "{"created":"@1640931725.659538000","description":"Error received from peer ipv4:10.96.23.132:19530","file":"src/core/lib/surface/call.cc","file_line":10...... (api_request.py:36)
```
3. when running `python scripts/hello_milvus.py --host 10.96.23.132`, insert raise error.
```
Error: <BaseException: (code=1, message=message send timeout: TimeoutError)>
Traceback (most recent call last):
  File "scripts/hello_milvus.py", line 117, in <module>
    hello_milvus(args.host)
  File "scripts/hello_milvus.py", line 49, in hello_milvus
    collection.insert(
  File "/Users/zilliz/opt/anaconda3/lib/python3.8/site-packages/pymilvus/orm/collection.py", line 543, in insert
    res = conn.bulk_insert(self._name, entities, partition_name, ids=None, timeout=timeout, **kwargs)
  File "/Users/zilliz/opt/anaconda3/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 84, in handler
    raise e
  File "/Users/zilliz/opt/anaconda3/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 76, in handler
    return func(*args, **kwargs)
  File "/Users/zilliz/opt/anaconda3/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 333, in bulk_insert
    raise err
  File "/Users/zilliz/opt/anaconda3/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 329, in bulk_insert
    raise BaseException(response.status.error_code, response.status.reason)
pymilvus.client.exceptions.BaseException: <BaseException: (code=1, message=message send timeout: TimeoutError)>
``` 

### Expected Behavior

all work well

### Steps To Reproduce

```markdown
cd `tests/python_client/chaos`
modify milvus tag in `cluster-values.yaml` to `master-20211230-95f0e9a`
run command `bash chaos_test.sh pulsar pod_kill chaos-test`
```

### Anything else?

log:
[pulsar-pod_kill-chaos-test-2021-12-31-14-22-58.zip](https://github.com/milvus-io/milvus/files/7795512/pulsar-pod_kill-chaos-test-2021-12-31-14-22-58.zip)

**Comment by bigsheeper** [comment]
```
[2021/12/31 06:10:14.117 +00:00] [DEBUG] [query_collection.go:602] ["get query message from unsolvedMsg"] [collectionID=430160143735586817] [msgID=430160114441037831] [reqTime_p=2021/12/31 06:09:44.439 +00:00] [serviceTime_p=2021/12/31 06:06:40.689 +00:00] [guaranteeTime_l=430160211984777217] [serviceTime_l=430160163815817217]
[2021/12/31 06:10:14.117 +00:00] [WARN] [query_collection.go:615] ["do query failed in doUnsolvedQueryMsg because timeout, collectionID =  430160143735586817 , msgID =  430160114441037831\n"]
[2021/12/31 06:10:14.117 +00:00] [DEBUG] [query_collection.go:602] ["get query message from unsolvedMsg"] [collectionID=430160145347772417] [msgID=430160114441037826] [reqTime_p=2021/12/31 06:09:39.312 +00:00] [serviceTime_p=2021/12/31 06:06:40.689 +00:00] [guaranteeTime_l=430160210640764928] [serviceTime_l=430160163815817217]
[2021/12/31 06:10:14.117 +00:00] [WARN] [query_collection.go:615] ["do query failed in doUnsolvedQueryMsg because timeout, collectionID =  430160145347772417 , msgID =  430160114441037826\n"]
```
search timeout, proxy's log shows that timeout_timestamp is `2021-12-31 14:10:04.429 +0800 CST`

**Comment by bigsheeper** [comment]
QueryNode reconnect to Pulsar from `2021-12-31T06:06:05` to `2021-12-31T06:09:57Z`

**Comment by zhuwenxing** [comment]
log:
[test-querynode-pod-kill-12-31-15-54-00.zip](https://github.com/milvus-io/milvus/files/7795629/test-querynode-pod-kill-12-31-15-54-00.zip)
```
(base) ➜  chaos git:(master) ✗ pytest -s -v ../testcases/test_e2e.py --host 10.96.23.132 --log-cli-level=INFO --capture=no
[get_env_variable] failed to get environment variables : 'CI_LOG_PATH', use default path : /tmp/ci_logs/
Test session starts (platform: darwin, Python 3.8.8, pytest 6.2.2, pytest-sugar 0.9.4)
cachedir: .pytest_cache
tags: all
metadata: {'Python': '3.8.8', 'Platform': 'macOS-10.16-x86_64-i386-64bit', 'Packages': {'pytest': '6.2.2', 'py': '1.10.0', 'pluggy': '0.13.1'}, 'Plugins': {'sugar': '0.9.4', 'repeat': '0.8.0', 'anyio': '2.2.0', 'rerunfailures': '9.1.1', 'assume': '2.4.3', 'tags': '1.7.2', 'timeout': '1.3.3', 'print': '0.2.1', 'html': '3.1.1', 'xdist': '2.2.1', 'allure-pytest': '2.7.0', 'metadata': '1.11.0', 'level': '0.1.1', 'dependency': '0.5.1', 'reportportal': '5.0.10', 'forked': '1.3.0', 'cov': '2.8.1'}}
rootdir: /Users/zilliz/workspace/backup/milvus/tests/python_client, configfile: pytest.ini
plugins: sugar-0.9.4, repeat-0.8.0, anyio-2.2.0, rerunfailures-9.1.1, assume-2.4.3, tags-1.7.2, timeout-1.3.3, print-0.2.1, html-3.1.1, xdist-2.2.1, allure-pytest-2.7.0, metadata-1.11.0, level-0.1.1, dependency-0.5.1, reportportal-5.0.10, forked-1.3.0, cov-2.8.1
selected 1 items

--------------------------------------------------------------------------------- live log setup ---------------------------------------------------------------------------------
[2021-12-31 14:56:17 - INFO - ci_test]: ################################################################################ (conftest.py:165)
[2021-12-31 14:56:17 - INFO - ci_test]: [initialize_milvus] Log cleaned up, start testing... (conftest.py:166)
[2021-12-31 14:56:17 - INFO - ci_test]: [setup_class] Start setup class... (client_base.py:45)
[2021-12-31 14:56:17 - INFO - ci_test]: *********************************** setup *********************************** (client_base.py:51)
[2021-12-31 14:56:17 - INFO - ci_test]: [setup_method] Start setup test case test_milvus_default. (client_base.py:52)
--------------------------------------------------------------------------------- live log call ----------------------------------------------------------------------------------
[2021-12-31 14:57:14 - INFO - ci_test]: assert create collection: 56.27781677246094, init_entities: 0 (test_e2e.py:24)
[2021-12-31 14:57:19 - INFO - ci_test]: assert insert: 4.9006781578063965 (test_e2e.py:31)
[2021-12-31 15:07:14 - INFO - ci_test]: assert flush: 595.6047179698944, entities: 3000 (test_e2e.py:39)
[2021-12-31 15:07:36 - ERROR - pymilvus.client.grpc_handler]: RPC error: _execute_search_requests{'API start': '2021-12-31 15:07:16.006312', 'RPC start': '2021-12-31 15:07:16.006330', 'RPC error': '2021-12-31 15:07:36.011353'} (grpc_handler.py:91)
[2021-12-31 15:07:36 - ERROR - pymilvus.client.grpc_handler]: RPC error: search{'API start': '2021-12-31 15:07:15.982011', 'RPC start': '2021-12-31 15:07:15.982021', 'RPC error': '2021-12-31 15:07:36.011893'} (grpc_handler.py:91)
[2021-12-31 15:07:36 - ERROR - ci_test]: Traceback (most recent call last):
  File "/Users/zilliz/workspace/backup/milvus/tests/python_client/utils/api_request.py", line 22, in inner_wrapper
    res = func(*args, **kwargs)
  File "/Users/zilliz/workspace/backup/milvus/tests/python_client/utils/api_request.py", line 56, in api_request
    return func(*arg, **kwargs)
  File "/Users/zilliz/opt/anaconda3/lib/python3.8/site-packages/pymilvus/orm/collection.py", line 696, in search
    res = conn.search(self._name, data, anns_field, param, limit, expr,
  File "/Users/zilliz/opt/anaconda3/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 92, in handler
    raise e
  File "/Users/zilliz/opt/anaconda3/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 76, in handler
    return func(*args, **kwargs)
  File "/Users/zilliz/opt/anaconda3/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 108, in handler
    return func(self, *args, **kwargs)
  File "/Users/zilliz/opt/anaconda3/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 467, in search
    return self._execute_search_requests(requests, timeout, **_kwargs)
  File "/Users/zilliz/opt/anaconda3/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 92, in handler
    raise e
  File "/Users/zilliz/opt/anaconda3/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 76, in handler
    return func(*args, **kwargs)
  File "/Users/zilliz/opt/anaconda3/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 417, in _execute_search_requests
    raise pre_err
  File "/Users/zilliz/opt/anaconda3/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 405, in _execute_search_requests
    response = ft.result()
  File "/Users/zilliz/opt/anaconda3/lib/python3.8/site-packages/grpc/_channel.py", line 744, in result
    raise self
grpc._channel._MultiThreadedRendezvous: <_MultiThreadedRendezvous of RPC that terminated with:
	status = StatusCode.DEADLINE_EXCEEDED
	details = "Deadline Exceeded"
	debug_error_string = "{"created":"@1640934456.010819000","description":"Error received from peer ipv4:10.96.23.132:19530","file":"src/core/lib/surface/call.cc","file_line":1068,"grpc_message":"Deadline Exceeded","grpc_status":4}"
>
 (api_request.py:35)
[2021-12-31 15:07:36 - ERROR - ci_test]: (api_response) : <_MultiThreadedRendezvous of RPC that terminated with:
	status = StatusCode.DEADLINE_EXCEEDED
	details = "Deadline Exceeded"
	debug_error_string = "{"created":"@1640934456.010819000","description":"Error received from peer ipv4:10.96.23.132:19530","file":"src/core/lib/surface/call.cc","file_line":10...... (api_request.py:36)
```
@bigsheeper 
This is the test that search timeout is set to 120s.

And another issue: **create collection: 56.27781677246094**
@czs007 Please take a look.

**Comment by zhuwenxing** [comment]
@sunby 
The flush time cost is always about 600s when running `testcases/test_e2e.py`.

**Comment by bigsheeper** [comment]
Proxy sent query message before QueryNode start. @zhuwenxing 

Proxy sent query message:
```
[2021/12/31 03:45:50.346 +00:00] [DEBUG] [task.go:2304] ["proxy sent one retrieveMsg"] [collectionID=430157940950564865] [msgID=430157906688386816] ["length of search msg"=1] [timeoutTs=430157953876885504]
[2021/12/31 03:45:50.346 +00:00] [DEBUG] [time_recorder.go:78] ["proxy execute query 430157906688386816: send retrieve request to message stream (3ms)"]
```

QueryNode start:
```
[2021/12/31 03:48:11.415 +00:00] [DEBUG] [service.go:263] ["QueryNode start done ..."]
[2021/12/31 03:48:11.415 +00:00] [DEBUG] [query_node.go:54] ["QueryNode successfully started"]
```

**Comment by bigsheeper** [comment]
> ```
> [2021/12/31 06:10:14.117 +00:00] [DEBUG] [query_collection.go:602] ["get query message from unsolvedMsg"] [collectionID=430160143735586817] [msgID=430160114441037831] [reqTime_p=2021/12/31 06:09:44.439 +00:00] [serviceTime_p=2021/12/31 06:06:40.689 +00:00] [guaranteeTime_l=430160211984777217] [serviceTime_l=430160163815817217]
> [2021/12/31 06:10:14.117 +00:00] [WARN] [query_collection.go:615] ["do query failed in doUnsolvedQueryMsg because timeout, collectionID =  430160143735586817 , msgID =  430160114441037831\n"]
> [2021/12/31 06:10:14.117 +00:00] [DEBUG] [query_collection.go:602] ["get query message from unsolvedMsg"] [collectionID=430160145347772417] [msgID=430160114441037826] [reqTime_p=2021/12/31 06:09:39.312 +00:00] [serviceTime_p=2021/12/31 06:06:40.689 +00:00] [guaranteeTime_l=430160210640764928] [serviceTime_l=430160163815817217]
> [2021/12/31 06:10:14.117 +00:00] [WARN] [query_collection.go:615] ["do query failed in doUnsolvedQueryMsg because timeout, collectionID =  430160145347772417 , msgID =  430160114441037826\n"]
> ```
> 
> search timeout, proxy's log shows that timeout_timestamp is `2021-12-31 14:10:04.429 +0800 CST`

QueryNode do query timeout because RootCoord sent ttMsg too slowly:
```
[2021/12/31 06:10:14.110 +00:00] [ERROR] [dml_channels.go:109] ["Broadcast failed"] [chanName=by-dev-rootcoord-dml_8] [stack="github.com/milvus-io/milvus/internal/rootcoord.(*dmlChannels).broadcast\n\t/go/src/github.com/milvus-io/milvus/internal/rootcoord/dml_channels.go:109\ngithub.com/milvus-io/milvus/internal/rootcoord.(*timetickSync).sendTimeTickToChannel\n\t/go/src/github.com/milvus-io/milvus/internal/rootcoord/timeticksync.go:368\ngithub.com/milvus-io/milvus/internal/rootcoord.(*timetickSync).startWatch.func1\n\t/go/src/github.com/milvus-io/milvus/internal/rootcoord/timeticksync.go:329"]
[2021/12/31 06:10:14.110 +00:00] [ERROR] [dml_channels.go:109] ["Broadcast failed"] [chanName=by-dev-rootcoord-dml_5] [stack="github.com/milvus-io/milvus/internal/rootcoord.(*dmlChannels).broadcast\n\t/go/src/github.com/milvus-io/milvus/internal/rootcoord/dml_channels.go:109\ngithub.com/milvus-io/milvus/internal/rootcoord.(*timetickSync).sendTimeTickToChannel\n\t/go/src/github.com/milvus-io/milvus/internal/rootcoord/timeticksync.go:368\ngithub.com/milvus-io/milvus/internal/rootcoord.(*timetickSync).startWatch.func1\n\t/go/src/github.com/milvus-io/milvus/internal/rootcoord/timeticksync.go:329"]
[2021/12/31 06:10:14.110 +00:00] [DEBUG] [timeticksync.go:330] ["SendTimeTickToChannel fail"] [error="message send timeout: TimeoutError"]
[2021/12/31 06:10:14.110 +00:00] [DEBUG] [timeticksync.go:330] ["SendTimeTickToChannel fail"] [error="message send timeout: TimeoutError"]
[2021/12/31 06:10:14.110 +00:00] [ERROR] [dml_channels.go:109] ["Broadcast failed"] [chanName=by-dev-rootcoord-dml_11] [stack="github.com/milvus-io/milvus/internal/rootcoord.(*dmlChannels).broadcast\n\t/go/src/github.com/milvus-io/milvus/internal/rootcoord/dml_channels.go:109\ngithub.com/milvus-io/milvus/internal/rootcoord.(*timetickSync).sendTimeTickToChannel\n\t/go/src/github.com/milvus-io/milvus/internal/rootcoord/timeticksync.go:368\ngithub.com/milvus-io/milvus/internal/rootcoord.(*timetickSync).startWatch.func1\n\t/go/src/github.com/milvus-io/milvus/internal/rootcoord/timeticksync.go:329"]
[2021/12/31 06:10:14.110 +00:00] [DEBUG] [timeticksync.go:330] ["SendTimeTickToChannel fail"] [error="message send timeout: TimeoutError"]
[2021/12/31 06:10:14.110 +00:00] [ERROR] [dml_channels.go:109] ["Broadcast failed"] [chanName=by-dev-rootcoord-dml_4] [stack="github.com/milvus-io/milvus/internal/rootcoord.(*dmlChannels).broadcast\n\t/go/src/github.com/milvus-io/milvus/internal/rootcoord/dml_channels.go:109\ngithub.com/milvus-io/milvus/internal/rootcoord.(*timetickSync).sendTimeTickToChannel\n\t/go/src/github.com/milvus-io/milvus/internal/rootcoord/timeticksync.go:368\ngithub.com/milvus-io/milvus/internal/rootcoord.(*timetickSync).startWatch.func1\n\t/go/src/github.com/milvus-io/milvus/internal/rootcoord/timeticksync.go:329"]
[2021/12/31 06:10:14.110 +00:00] [DEBUG] [timeticksync.go:330] ["SendTimeTickToChannel fail"] [error="message send timeout: TimeoutError"]
[2021/12/31 06:10:14.110 +00:00] [WARN] [timeticksync.go:339] ["rootcoord send tt to all channels too slowly"] [chanNum=14] [span=30000]
```

**Comment by xiaofan-luan** [comment]
@bigsheeper any reason why the send is slow?

**Comment by bigsheeper** [comment]
> @bigsheeper any reason why the send is slow?

RootCoord keep throwing TimeoutError, it seems that RootCoord doesn't connect to Pulsar correctly after Pulsar restarting, I'm trying to figure out why.

**Comment by bigsheeper** [comment]
Cannot reproduce with version:master-20211230-95f0e9a, chaos test blocked by flush issue #14300

**Comment by zhuwenxing** [comment]
Not reproduced in the test as following

* deploy Milvus
* `cd  tests/python_client/deploy/scripts` and `python action_before_reinstall.py`
* inject chaos for 2 mins and then delete it
* `cd  tests/python_client/deploy/scripts` and `python action_after_reinstall.py`

The difference is during the chaos, no requests are sent to Milvus.

**Comment by xiaofan-luan** [comment]
> Not reproduced in the test as following
> 
> * deploy Milvus
> * `cd  tests/python_client/deploy/scripts` and `python action_before_reinstall.py`
> * inject chaos for 2 mins and then delete it
> * `cd  tests/python_client/deploy/scripts` and `python action_after_reinstall.py`
> 
> The difference is during the chaos, no requests are sent to Milvus.

Do we try to reproduce with request?

**Comment by zhuwenxing** [comment]
@bigsheeper Hi, the issue reproduced in this job https://github.com/milvus-io/milvus/runs/4754495386?check_suite_focus=true.
error in client
```
Addr [127.0.0.1:19530] _execute_search_requests
RPC error: <_MultiThreadedRendezvous of RPC that terminated with:
	status = StatusCode.DEADLINE_EXCEEDED
	details = "Deadline Exceeded"
	debug_error_string = "{"created":"@1641754326.755208283","description":"Error received from peer ipv4:127.0.0.1:19530","file":"src/core/lib/surface/call.cc","file_line":1067,"grpc_message":"Deadline Exceeded","grpc_status":4}"
>
	{'API start': '2022-01-09 18:50:06.754579', 'RPC start': '2022-01-09 18:50:06.754859', 'RPC error': '2022-01-09 18:52:06.755595'}

Addr [127.0.0.1:19530] search
RPC error: <_MultiThreadedRendezvous of RPC that terminated with:
	status = StatusCode.DEADLINE_EXCEEDED
	details = "Deadline Exceeded"
	debug_error_string = "{"created":"@1641754326.755208283","description":"Error received from peer ipv4:127.0.0.1:19530","file":"src/core/lib/surface/call.cc","file_line":1067,"grpc_message":"Deadline Exceeded","grpc_status":4}"
>
	{'API start': '2022-01-09 18:50:06.732128', 'RPC start': '2022-01-09 18:50:06.732363', 'RPC error': '2022-01-09 18:52:06.755834'}
```
error in server proxy
```
[2022/01/09 18:58:12.942 +00:00] [ERROR] [task_scheduler.go:477] ["Failed to post-execute task: searchTask:wait to finish failed, timeout: 430375896088885295"] [traceID=473546239de590a5] [stack="github.com/milvus-io/milvus/internal/proxy.(*taskScheduler).processTask\n\t/go/src/github.com/milvus-io/milvus/internal/proxy/task_scheduler.go:477"]
[2022/01/09 18:58:12.942 +00:00] [WARN] [impl.go:2204] ["Search failed to WaitToFinish"] [error="proxy TaskCondition context Done"] [traceID=473546239de590a5] [role=proxy] [msgID=430375896088885295] [db=] [collection=hello_milvus] [partitions="[]"] [dsl="count > 100"] [len(PlaceholderGroup)=1039] [OutputFields="[count,random_value]"] [search_params="[{\"key\":\"anns_field\",\"value\":\"float_vector\"},{\"key\":\"topk\",\"value\":\"5\"},{\"key\":\"metric_type\",\"value\":\"L2\"},{\"key\":\"params\",\"value\":\"{\\\"nprobe\\\":10}\"},{\"key\":\"round_decimal\",\"value\":\"-1\"}]"] [travel_timestamp=0] [guarantee_timestamp=0]
[2022/01/09 18:58:13.948 +00:00] [DEBUG] [impl.go:3363] [RegisterLink] [role=proxy] ["state code of proxy"=Healthy]
```

**Comment by bigsheeper** [comment]
QueryNode raised `timeoutError` when producing search result after Pulsar restart. It looks like the same reason of `timeoutError` in RootCoord which leed to QueryNode search timeout and DataNode flush timeout（#14300）.
```
[2022/01/09 18:56:43.202 +00:00] [ERROR] [query_collection.go:1388] ["message send timeout: TimeoutError"] [stack="github.com/milvus-io/milvus/internal/querynode.(*queryCollection).publishQueryResult\n\t/go/src/github.com/milvus-io/milvus/internal/querynode/query_collection.go:1388\ngithub.com/milvus-io/milvus/internal/querynode.(*queryCollection).search\n\t/go/src/github.com/milvus-io/milvus/internal/querynode/query_collection.go:1199\ngithub.com/milvus-io/milvus/internal/querynode.(*queryCollection).doUnsolvedQueryMsg\n\t/go/src/github.com/milvus-io/milvus/internal/querynode/query_collection.go:652"]
[2022/01/09 18:56:43.202 +00:00] [WARN] [query_collection.go:660] ["message send timeout: TimeoutError"]
[2022/01/09 18:57:13.203 +00:00] [WARN] [query_collection.go:663] ["message send timeout: TimeoutError"]
[2022/01/09 18:57:13.203 +00:00] [DEBUG] [query_collection.go:672] ["do query done in doUnsolvedMsg"] [collectionID=430375896089038145] [msgID=430375896088885295]
```

**Comment by xige-16** [comment]
https://github.com/apache/pulsar-client-go/pull/703 seems has fixed the issue

**Comment by sunby** [comment]
@xige-16 Can we use the fixed commit as our pulsar client dependency version?

**Comment by xige-16** [comment]
> 

Need to assess the risk, after all, it is not a release version
